From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 12 May 2020 10:39:58 +0200
Subject: [PATCH] imx8mm-var-dart-nrt: Enable M4, run from DDR

Enable the M4 core, set firmware binary name,
set the DTB and let the binary reside in the DDR.

We do so as DDR was requested for rpmsg. At least
3GB RAM are necessary to allow the M4 to access
the RPMSG area according to the Variscite docs.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/imx8mm_var_dart.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index deb86abfb7..e51c2cd7c3 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -115,10 +115,10 @@
 	"mmcblk=1\0" \
 	"mmcautodetect=yes\0" \
 	"mmcpart=1\0" \
-	"m4_addr=0x7e0000\0" \
-	"m4_bin=hello_world.bin\0" \
-	"use_m4=no\0" \
-	"loadm4bin=load mmc ${mmcdev}:${mmcpart} ${m4_addr} ${bootdir}/${m4_bin}\0" \
+	"m4_addr=0x7e000000\0" \
+	"m4_bin=m4_fw.bin\0" \
+	"use_m4=yes\0" \
+	"loadm4bin=load mmc ${mmcdev}:${resin_boot_part} ${m4_addr} ${m4_bin}\0" \
 	"runm4bin=" \
 		"if test ${m4_addr} = 0x7e0000; then " \
 			"echo Booting M4 from TCM; " \
@@ -144,7 +144,7 @@
 					"setenv fdt_file fsl-imx8mm-var-som.dtb; " \
 				"fi;" \
 			"else " \
-				"setenv fdt_file fsl-imx8mm-var-dart.dtb;" \
+				"setenv fdt_file fsl-imx8mm-var-dart-m4.dtb;" \
 			"fi; " \
 		"fi; \0" \
 	"loadfdt=run findfdt; " \
-- 
2.17.1

